components/brave_wallet_ui/common/slices/api.slice.ts

```diff
@@ -1279,12 +1297,65 @@ export function createWalletApi () {
             : (balancesResult &&
                 Object.keys(balancesResult).map((tokenAddress) => ({
                   type: 'TokenBalancesForChainId',
-                  id:
-                    arg.coin === CoinTypes.ETH
-                      ? `${arg.address}-${arg.coin}-${arg.chainId}-${tokenAddress}`
-                      : `${arg.pubkey}-${arg.coin}-${arg.chainId}-${tokenAddress}`
+                  id: [arg.accountId.uniqueKey, arg.chainId, tokenAddress].join('-')
                 }))) || ['TokenBalancesForChainId']
       }),
+      getHardwareAccountDiscoveryBalance: query<
```

Comment:
is this needed? could we instead fetch the same results my passing the native asset to `getAccountTokenCurrentBalance` (like we do in `usePendingTransaction`) ?

Reply:
I'm gradually switching from referencing account by address to referencing account by accountId. This is needed for bitcoin where accounts do not have 'address'.
For this particular case getAccountTokenCurrentBalance now accepts AccountId and it becomes unusable for hardware accounts discovery when we don't associated accounts yet. I'm thinking of getting native balance of possible hardware account as a special quite rare case which is ok to have separate endpoint.
The idea for future is to have two separate backend endpoints like
`getAccountTokenCurrentBalance(accountId, token)`
`getHardwareAccountDiscoveryBalance(coin, address)`
which will make much of code currently siding in react frontend also reusable for android and ios.